### DISCUSSION TITLE
Conditional Polymorphism

### DISCUSSION THREAD

**Discussion by A-Corregidor** [loopbackio/loopback-next#9017] [Q&A]
Hi there!
I can't find on docs if it's possible to build a conditional polymorphism, omitting it when the discriminator is null.

Let's suppose model A:

```
{
 id: string,
 entityType: string | null,
 entityId: string | null
}

```

Thinking of 3 different scenariors:
- entityType corresponds to another model, entityId corresponds with other model's id -> Everything is OK
- entityType corresponds to another model, entityId is null -> I'm not sure when would this happen, but everything is OK
- entityType is null, entityId is null -> Error " Invalid class name null by discriminator natureType. Please check polymorphic types and the discriminator"

Is there any way to ignore polymorphism when discriminator is null?

**Comment by A-Corregidor** [comment]
Up

**Comment by achrinza** [comment]
Hi @A-Corregidor, I understand some time has passed since this discussion was initiated. Unfortunately I'm not aware of this being possible with the current implementation. Polymorphism is still a relatively-new feature in LoopBack 4, so there's still plenty of opportunity to improve on it.
